File: dataset/Vanilla/GEMINI_gemini-2.5-pro/Task100.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.regex.Matcher;
import java.util.regex.Pattern;
import java.util.regex.PatternSyntaxException;
import java.util.stream.Stream;

public class Task100 {

    /**
     * Searches for a regex pattern within a file and prints matching lines.
     * @param regexPattern The regular expression pattern to search for.
     * @param fileName The path to the file to search in.
     */
    public static void searchInFile(String regexPattern, String fileName) {
        System.out.println("Searching for pattern '" + regexPattern + "' in file '" + fileName + "'...");
        try {
            Pattern pattern = Pattern.compile(regexPattern);
            try (Stream<String> lines = Files.lines(Paths.get(fileName))) {
                lines.forEach(line -> {
                    Matcher matcher = pattern.matcher(line);
                    if (matcher.find()) {
                        System.out.println("Found match: " + line);
                    }
                });
            } catch (IOException e) {
                System.err.println("Error reading file: " + e.getMessage());
            }
        } catch (PatternSyntaxException e) {
            System.err.println("Invalid regular expression: " + e.getMessage());
        }
        System.out.println("------------------------------------");
    }

    /**
     * Sets up test data and runs test cases.
     */
    public static void runTestCases() {
        String testFileName = "test_file_java.txt";
        try (FileWriter writer = new FileWriter(testFileName)) {
            writer.write("Hello world, this is a test file.\n");
            writer.write("The year is 2024.\n");
            writer.write("Contact us at test@example.com for more info.\n");
            writer.write("Or call 123-456-7890.\n");
            writer.write("This line contains no special characters or numbers\n");
            writer.write("Another email: another.email@domain.org.\n");
        } catch (IOException e) {
            System.err.println("Failed to create test file: " + e.getMessage());
            return;
        }

        // Test Case 1: Match a 4-digit number
        searchInFile("\\d{4}", testFileName);

        // Test Case 2: Match an email address
        searchInFile("\\w+@\\w+\\.\\w+", testFileName);

        // Test Case 3: Match a line starting with "Hello"
        searchInFile("^Hello", testFileName);

        // Test Case 4: Match a line ending with "numbers"
        searchInFile("numbers$", testFileName);

        // Test Case 5: A pattern that won't match anything
        searchInFile("NonExistentPatternXYZ", testFileName);

        // Cleanup the test file
        new File(testFileName).delete();
    }

    public static void main(String[] args) {
        if (args.length == 2) {
            // Use command-line arguments
            String regexPattern = args[0];
            String fileName = args[1];
            searchInFile(regexPattern, fileName);
        } else {
            // Run built-in test cases
            System.out.println("Usage: java Task100 <regex_pattern> <file_name>");
            System.out.println("Running test cases instead...\n");
            runTestCases();
        }
    }
}

File: dataset/CoT/GEMINI_gemini-2.5-pro/Task100.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <regex.h>

/**
 * @brief Searches a file for lines matching a given POSIX regular expression.
 * 
 * @param regexPattern The regular expression pattern string.
 * @param fileName The path to the file to be searched.
 */
void searchFileWithRegex(const char* regexPattern, const char* fileName) {
    regex_t regex;
    int reti;
    char msgbuf[100];

    // Step 1: Compile the regular expression
    reti = regcomp(&regex, regexPattern, REG_EXTENDED);
    if (reti) {
        regerror(reti, &regex, msgbuf, sizeof(msgbuf));
        fprintf(stderr, "Error: Regex compilation failed: %s\n", msgbuf);
        return;
    }

    // Step 2: Open the file
    FILE* fp = fopen(fileName, "r");
    if (fp == NULL) {
        fprintf(stderr, "Error: Could not open file: %s\n", fileName);
        regfree(&regex); // Free regex memory on error
        return;
    }

    // Step 3: Read file line by line and search for matches
    char* line = NULL;
    size_t len = 0;
    ssize_t read;
    int lineNumber = 1;
    int found = 0;

    while ((read = getline(&line, &len, fp)) != -1) {
        // Remove trailing newline character if it exists
        if (line[read - 1] == '\n') {
            line[read - 1] = '\0';
        }
        
        reti = regexec(&regex, line, 0, NULL, 0);
        if (!reti) { // 0 means a match was found
            printf("Match found on line %d: %s\n", lineNumber, line);
            found = 1;
        } else if (reti != REG_NOMATCH) {
            regerror(reti, &regex, msgbuf, sizeof(msgbuf));
            fprintf(stderr, "Regex match failed on line %d: %s\n", lineNumber, msgbuf);
        }
        lineNumber++;
    }

    if (!found) {
        printf("No matches found for pattern '%s' in file '%s'.\n", regexPattern, fileName);
    }

    // Step 4: Clean up resources
    fclose(fp);
    if (line) {
        free(line);
    }
    regfree(&regex);
}

void runTests() {
    const char* testFileName = "test_file.txt";
    // Create a temporary test file
    FILE* fp = fopen(testFileName, "w");
    if (fp == NULL) {
        fprintf(stderr, "Failed to create test file.\n");
        return;
    }
    fprintf(fp, "Hello world, this is a test file.\n");
    fprintf(fp, "The quick brown fox jumps over the lazy dog.\n");
    fprintf(fp, "123-456-7890 is a phone number.\n");
    fprintf(fp, "Another line with the word world.\n");
    fprintf(fp, "Email: test@example.com\n");
    fprintf(fp, "invalid-email@.com\n");
    fclose(fp);

    // Note: C POSIX regex uses a slightly different syntax
    const char* testCases[][2] = {
        {"world", "Test Case 1: Simple word match"},
        {"^[A-Z]", "Test Case 2: Match lines starting with a capital letter"},
        {"[0-9]{3}-[0-9]{3}-[0-9]{4}", "Test Case 3: Match a phone number format"},
        {"(fox|dog)", "Test Case 4: Match using alternation"},
        {"[a-zA-Z0-9._%%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}", "Test Case 5: Match a valid email address"}
    };
    int numTests = sizeof(testCases) / sizeof(testCases[0]);

    for (int i = 0; i < numTests; i++) {
        const char* pattern = testCases[i][0];
        const char* description = testCases[i][1];
        printf("\n--- %s ---\n", description);
        printf("Pattern: %s\n", pattern);
        printf("------------------------------------\n");
        searchFileWithRegex(pattern, testFileName);
    }

    // Clean up the test file
    remove(testFileName);
}

int main(int argc, char* argv[]) {
    // Check if command line arguments are provided
    if (argc == 3) {
        printf("--- Running with Command Line Arguments ---\n");
        const char* regex = argv[1];
        const char* file = argv[2];
        printf("Pattern: %s\n", regex);
        printf("File: %s\n", file);
        printf("-------------------------------------------\n");
        searchFileWithRegex(regex, file);
    } else {
        printf("Usage: %s <regex_pattern> <file_name>\n", argv[0]);
        printf("Running built-in test cases as no arguments were provided.\n");
        runTests();
    }
    return 0;
}